Datastore/tests/Unit/EntityMapperTest.php

```diff
@@ -898,4 +917,52 @@ public function testObjectPropertyInt64()
             'integerValue' => $int64->get()
         ], $res);
     }
+
+    public function datastoreToSimpleDoubleValueCases()
+    {
+        return [
+            [1.1, 1.1],
+
+            // Happens when using rest client
+            ['Infinity', INF],
+            ['-Infinity', -INF],
+            ['NaN', NAN],
+
+            // Happens when using grpc client
+            [INF, INF],
+            [-INF, -INF],
+            [NAN, NAN]
+        ];
+    }
+
+    public function valueObjectDoubleCases()
+    {
+        return [
+            [INF, INF],
+            [1.1, 1.1],
+            [-INF, -INF],
+            [NAN, NAN]
+        ];
+    }
+
+    public function valueObjectDoubleForRestCases()
+    {
+        return [
+            [INF, 'Infinity'],
+            [1.1, 1.1],
+            [-INF, '-Infinity'],
+            [NAN, 'NaN']
+        ];
+    }
+
+    private function compareResult($result, $expected)
```

Comment:
Change this to `compareResult($expected, $return)` to match the phpunit's assert parameters pattern.
Also check whether the usage is appropriate, since only `$result` is asserted.

Reply:
Ah yes, nice catch. Updated.